src/tpt_reports/tpt_reports.py

```diff
@@ -45,8 +52,85 @@ def load_json_file(file_path):
         return None
 
 
-# Issue #4 - Add ReportLab code and library
-# TODO: Add in the ReportLab code, library, parsing logic and remove this comment.
+# Issue #21 - test parse_json()
+# TODO: Add unit tests for following logic and remove this comment.
+def parse_json(data):
+    """Parse JSON object for values to report."""
+    payloads_list = []
+    payloads_meta = {}
+    num_payloads = 0
+    host_blocked = 0
+    host_not_blocked = 0
+    border_blocked = 0
+    border_not_blocked = 0
+    try:
+        if data:
+            for payload in data["payloads"]:
+                num_payloads += 1
+                payload_data = {}
+                payload_data["Payload"] = payload["payload_description"]
+                # c2_protocol requires additional space to match input file format
+                payload_data["C2 Protocol"] = payload["c2_protocol "]
+
+                if payload["border_protection"] == "N":
+                    payload_data["Border Protection"] = "Not Blocked"
+                    border_not_blocked += 1
+                elif payload["border_protection"] == "B":
+                    payload_data["Border Protection"] = "Blocked"
+                    border_blocked += 1
+                else:
+                    raise ValueError("border_protection value must be either B or N")
+
+                if payload["host_protection"] == "N":
+                    payload_data["Host Protection"] = "Not Blocked"
+                    host_not_blocked += 1
+                elif payload["host_protection"] == "B":
+                    payload_data["Host Protection"] = "Blocked"
+                    host_blocked += 1
+                else:
+                    raise ValueError("host_protection value must be either B or N")
+
+                payloads_list.append(payload_data)
+
+        payloads_meta["num_payloads"] = num_payloads
+        payloads_meta["host_blocked"] = host_blocked
+        payloads_meta["host_not_blocked"] = host_not_blocked
+        payloads_meta["border_blocked"] = border_blocked
+        payloads_meta["border_not_blocked"] = border_not_blocked
+        payloads_meta["num_blocked"] = num_payloads
```

Comment:
Why is `payloads_meta["num_blocked"] = num_payloads`?  

If this is correct, I'd definitely add an explanatory comment because it's confusing since you also have `payloads_meta["num_payloads"] = num_payloads`.

If this is not correct, well, you know what to do. 😄

Reply:
For future reference, it's best to not include line numbers in commit messages, since line numbers change over time as code is added/deleted.  Also, if someone is reading through the log of commit messages, line numbers don't provide much useful information.  A better practice is to specify what was changed and why, rather than mention line numbers.